File: src/gestorAplicacion/lugares/CampoDeBatalla.java
package gestorAplicacion.lugares;

import gestorAplicacion.pokemon.*;
import java.util.ArrayList;
import java.util.Scanner;
import java.io.*;

public class CampoDeBatalla {
	int tiempoEspera=1500;

    int turno=0;
    public void resetTurno(){
        turno=1;
    }
    public int getturno(){
        return turno;
    }
    public void seguirturno(){
        turno++;
    }
    public void ComenzarBatalla(Pokemon p1,Pokemon p2) {
        Scanner teclado = new Scanner(System.in);

    }
    public void tespe(){
        try {
            Thread.sleep(tiempoEspera);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
    }
    public String verificarvidas(Pokemon p1,Pokemon p2){
        if (p1.getVida()==0 || p2.getVida()==0){
            String[] a1 = new String[1];

            if (p1.getVida()==0){
                a1[0]=p2.getNombre();
            }
            else if (p2.getVida()==0){
                a1[0]=p1.getNombre();
            }

            return "\nHA TERMINADO LA PELEA\n El ganador ha sido "+a1[0]+"\n";
        }
        else{
            return "";

        }
    }






}

File: python/gestorAplicacion/lugares/CampoDeBatalla.py
from gestorAplicacion.pokemon.Pokemon import Pokemon

class CampoDeBatalla():
    
    _tiempoEspera = 1500
    _turno = 0

    #metodo contructor vacio
    def __init__(self) -> None:
        pass

    """metodos para estructura a la batalla"""

    def resetTurno(self):
        CampoDeBatalla._turno = 1

    def getturno(self):
        return CampoDeBatalla._turno

    def seguirturno(self):
        CampoDeBatalla._turno += 1
    
    def ComenzarBatalla(self, pokemon1, pokemon2):
        accion = input()

    def tespe(self):
        pass        #no se que se hace en este metodo

    def verificarvidas(self, pokemon1, pokemon2):
        if (pokemon1.getVida() == 0) or (pokemon2.getVida() == 0):
            ganador = ""
            if pokemon1.getVida() == 0:
                ganador = pokemon1.getNombre()
            elif pokemon2.getVida() == 0:
                ganador = pokemon2.getNombre()
            return "HA TERMINADO LA PELEA\nEL GANADOR HA SIDO " + ganador
        else:
            return ""


